### ISSUE TITLE
DISCUSSION: What open datasets do you want in Pelias/Mapzen Search?

### ISSUE THREAD

**Issue by riordan** [pelias/pelias#254]
Pelias (/Mapzen Search) can only be as good as the data available to it. And so far, we've done a lot to make the best use of the best collaborative open data projects we can find. But there are plenty of other sources of data that have the potential to open geocoding up to parts of the world where we don't yet offer it.

We suspect you probably know some of the best sources already that would make Pelias/Mapzen Search substantially better. Please, leave us comment below with a particular target dataset that you'd like to see Pelias/Mapzen Search rely upon.

We'd love to see new sources of:
- Address data (so we can provide it to OpenAddresses)
- Road network data for interpolation
- Postcodes
- Venues and named points of interest

This is why we think open will win: when people can share what we know, knowledge grows faster and we can build so much more.

**Comment by riordan** [comment]
For reference here's the datasets we currently use:
- [OpenStreetMap](https://openstreetmap.org) - Streets, venues, addresses, areas of interest
- [OpenAddresses](http://openaddresses.io) - addresses
- [Quattroshapes](http://quattroshapes.com) - named administrative places (soon to be deprecated)
- [Who's on First](https://whosonfirst.mapzen.com) - named administrative places, neighborhoods, airports (polygons)
- [Geonames](http://geonames) - named places (points)

**Comment by riordan** [comment]
Also here are a few open sources we're tracking now. No promises if/when they'll all make it into the service (and if it makes sense to route them through other, broader projects, like [OpenAddresses](http://openaddresses.io) or [Who's on First](https://whosonfirst.mapzen.com)), but we want to be transparent in what we've been thinking about.

- US ZIP Codes | [TIGER/Lines](https://www.census.gov/geo/maps-data/data/tiger-line.html) | Polygons
- UK Postal Codes | [Ordinance Survey Code Point Open](https://www.ordnancesurvey.co.uk/business-and-government/products/code-point-open.html) | Points
- SimpleGeo Venues | [Who's on First Venues](https://github.com/whosonfirst/whosonfirst-venue) | Points
- US Address Interpolation | [TIGER/Lines](https://www.census.gov/geo/maps-data/data/tiger-line.html) | Road interpolation lines
- Canada Address Interpolation | [Statistics Canada Road Network Data](https://www12.statcan.gc.ca/census-recensement/2011/geo/RNF-FRR/index-eng.cfm) | Road interpolation lines

**Comment by mtmail** [comment]
It's GB postal codes. The Northern Ireland postcodes are managed by https://www.dfpni.gov.uk/publications/lps-copyright-licensing-digital-application-forms (kind of a separate Ordinance Survey). Those are reselling Royalmail data and require a commercial license. I know, pretty annoying, especially since you can download the NI postcodes just fine.

For Canadian postcodes watch http://geocoder.ca/?sued=1#updates (and/or add to their legal defense fund)

**Comment by riordan** [comment]
Totally, thank you for catching the NI issue, @mtmail.

And we'll keep our eyes on geocoder.ca.

**Comment by orangejulius** [comment]
I would like to propose that we build an importer for [All The Places](https://www.alltheplaces.xyz/). cc @iandees

**Comment by hughcameron** [comment]
Hi @orangejulius, I noticed your [Docker project for Australia](https://github.com/pelias/docker/tree/master/projects/australia). The Australian Government also makes a [Geocoded National Address File](https://data.gov.au/dataset/geocoded-national-address-file-g-naf) each quarter which may be useful to develop an importer for.

**Comment by trescube** [comment]
Hey @hughcameron, Pelias uses [openaddresses](https://openaddresses.io/) which contains that curated [Australia-wide](https://github.com/openaddresses/openaddresses/blob/master/sources/au/countrywide.json) source, kept up-to-date by the inimitable @andrewharvey.

**Comment by orangejulius** [comment]
@hughcameron, yup, what @trescube said. The excellent coverage via G-NAF data is part of the reason I made the Australia Docker project. We plan to use Australia as a mid-sized build for testing pelias going forward because we know there is great address coverage. If you have a chance to try it out let me know!

**Comment by hughcameron** [comment]
Hey @orangejulius & @trescube - thanks for letting me know. I got the Australia project up and running earlier in the week using forward geocoding for validation and parsing - really powerful stuff. Great work!

**Comment by yopaseopor** [comment]
I suggest two or three public sources:
Adresses. Directorate General for Cadastre. Spain[]( http://www.catastro.minhap.es/webinspire/index.html)
Institut Geològic i Cartogràfic de Catalunya. Name of streets and housenumbers [](http://www.icgc.cat/Descarregues/Llocs/WMS-Noms-i-numeracio-de-carrers)
Barcelona city: all the adresses and housenumbers: [](https://opendata-ajuntament.barcelona.cat/data/ca/dataset/taula-direle)

**Comment by Joxit** [comment]
I think [openaddresses](https://github.com/openaddresses/openaddresses) will be happy to have this source : https://opendata-ajuntament.barcelona.cat/data/ca/dataset/taula-direle

**Comment by trescube** [comment]
Feel free to open a PR!

**Comment by pravink** [comment]
Just wanted to know If I can import locations with private metadata on it as a custom datasource for my own Pelias instance? so that i can search those using Pelias API like search/autocomplete?

**Comment by vincentfenet** [comment]
In france, you have the official https://adresse.data.gouv.fr/ too

**Comment by Joxit** [comment]
`adresse.data.gouv.fr` is already in [OpenAddresses](https://github.com/openaddresses/openaddresses/blob/master/sources/fr/countrywide.json) thanks to a member of Etalab :smile:

**Comment by hennihaus** [comment]
It would be nice to an official importer for gtfs-data.

**Comment by gelsas** [comment]
I noticed that the Canadian Postcodes are not really good in pelias yet.

This company is making monthly updated canadian zip code data (890.000 entries right now) avaiable for free: https://www.serviceobjects.com/blog/unique-us-canadian-zip-code-files-available-download/

This would make Canadian address data much more accurate.

**Comment by orangejulius** [comment]
Hi @gelsas,
Thanks for the recommendation. Open data for Canadian postcodes is hard to come by. The copyright over the data is held by Canada Post and they have [defended that copyright in the past](https://gogeomatics.ca/postal-codes-canada-post-backs-down-on-crowdsourced-open-data/), as our fellow geocodeur Ervin from geocode.xyx knows well.

It appears that the [license](https://www.serviceobjects.com/zip-eula) for that zip file from ServiceObjects is effectively proprietary, and certainly not suitable for inclusion by default in Pelias. It's likely anyone using Pelias for their own needs wouldn't want to use it either.

**Comment by gelsas** [comment]
Is this included in pelias ?
Australia:
https://data.gov.au/dataset/geocoded-national-address-file-g-naf

**Comment by gelsas** [comment]
Is this data sets already used in pelias:
iceland:
https://github.com/sveinbjornt/iceaddr

Is NZ data included already?
https://data.linz.govt.nz/

**Comment by orangejulius** [comment]
Hi @gelsas,

The G-NAF dataset from Australia, New Zealand's countrywide address dataset (and several others specific to particular cities or regions), and the Islandic address registry are all supported out of the box by Pelias through the wonderful [OpenAddresses](https://openaddresses.io/) project.

The OpenAddresses folks are pretty diligent about adding new public, appropriately licensed address datasets, so they'll generally have anything like that. You can always check in their [sources list](https://github.com/openaddresses/openaddresses/tree/master/sources) and if you do happen to find one that's missing they would definitely want to know :)
